src/nwb/NWBFile.cpp

```diff
@@ -172,6 +170,43 @@ Status NWBFile::createFileStructure(const std::string& identifierText,
   return Status::Success;
 }
 
+Status NWBFile::createElectrodesTable(
+    std::vector<Types::ChannelVector> recordingArrays)
+{
+  std::unique_ptr<NWB::ElectrodeTable> electrodeTable =
+      std::make_unique<NWB::ElectrodeTable>(m_io);
+  electrodeTable->initialize();
+  for (const auto& channelVector : recordingArrays) {
+    electrodeTable->addElectrodes(channelVector);
+  }
+
+  for (size_t i = 0; i < recordingArrays.size(); ++i) {
+    const auto& channelVector = recordingArrays[i];
+
+    // Setup electrodes and devices
+    std::string groupName = channelVector[0].getGroupName();
+    std::string devicePath = AQNWB::mergePaths("/general/devices", groupName);
+    std::string electrodePath =
+        AQNWB::mergePaths("/general/extracellular_ephys", groupName);
+
+    // Check if device exists for groupName, create device and electrode group
+    // if it does not
+    if (!m_io->objectExists(devicePath)) {
+      NWB::Device device = NWB::Device(devicePath, m_io);
+      device.initialize("description", "unknown");
+
+      NWB::ElectrodeGroup elecGroup = NWB::ElectrodeGroup(electrodePath, m_io);
+      elecGroup.initialize("description", "unknown", device);
+    }
+  }
+
+  // write electrodes information to datasets
+  // (requires that ElectrodeGroup data is written)
```

Comment:
If the ElectrodeGroup data must be written, should this function then call ``m_io->flush()`` here first?